yt/data_objects/region_expression.py

```diff
@@ -133,3 +147,31 @@ def _create_region(self, bounds_tuple):
         if all(d is not None for d in dims):
             return self.ds.arbitrary_grid(left_edge, right_edge, dims)
         return self.ds.region(center, left_edge, right_edge)
+
+    def _create_point(self, point_tuple):
+        coord = [self._spec_to_value(p) for p in point_tuple]
+        return self.ds.point(coord)
+
+    def _create_ray(self, ray_slice):
+        start_point = [self._spec_to_value(v) for v in ray_slice.start]
+        end_point = [self._spec_to_value(v) for v in ray_slice.stop]
+        return self.ds.ray(start_point, end_point)
+
+    def _create_ortho_ray(self, ray_tuple):
+        axis = None
+        new_slice = []
+        coord = []
+        for ax, v in enumerate(ray_tuple):
+            if not isinstance(v, slice):
+                coord.append(self._spec_to_value(v))
+                new_slice.append(slice(None, None, None))
+            else:
+                if axis is not None: raise RuntimeError
+                axis = ax
+                new_slice.append(v)
+        if axis == 1: coord = [coord[1], coord[0]]
+        source = self._create_region(new_slice)
+        ray = self.ds.ortho_ray(axis, coord, data_source=source)
+        if getattr(new_slice[axis].step, "imag", 0.0) != 0.0:
```

Comment:
this makes me think we should implement a sampling function at some point so that `imag` on a ray is interesting ...